Source/Input/ConfigManager/ConfigManager.cpp

```diff
@@ -0,0 +1,134 @@
+#include "ConfigManager.h"
+
+
+bool LogCategory::IfElementExists(std::string keyValue) {
+
+	if (params.find(keyValue) != params.end()) {
+		return true;
+	}
+
+	return false;
+}
+
+std::string LogCategory::getValue(std::string keyValue) {
+
+
+	if (IfElementExists(keyValue)) {
+
+		return params[keyValue];
+	}
+
+	std::cerr << "Key value not found" << std::endl;
+	system("PAUSE");
+	exit(1);
+}
+
+void LogCategory::addNewParam(std::pair <std::string, std::string> param) {
+
+	params.insert(param);
+}
+
+/*
+----------------------------------------------------------------------------
+*/
+
+ConfigManager* ConfigManager::create(std::string fileName) {
+	ConfigManager* cm = new ConfigManager();
+	cm->readInputFile(fileName);
+
+	return cm;
+}
+
+void ConfigManager::readInputFile(std::string fileName) {
+
+	inFile.open(fileName);
+
+	if (inFile.is_open()) {
+		createCategories();
+		inFile.close();
+	}
+	else {
+		std::cerr << "Cannot open file" << std::endl;
+		system("PAUSE");
+		exit(1);
+	}
+
+}
+
+void ConfigManager::createCategories() {
+
+	std::string line;
+	std::string currentCategory;
+
+	while (!inFile.eof()) {
+
+		std::getline(inFile, line);
+
+		if (line[0] == '[') {
+			std::ostringstream oss;
+
+			for (int i = 1; i < line.size() - 1, line[i] != ']'; ++i) {
```

Comment:
` i < line.size() - 1, line[i] != ']'`
what's with that condition? use some binary logic op